product/cmd/grpc/server/server_test.go

```diff
@@ -0,0 +1,117 @@
+// Package main is the implementation of the grpc product service server.
+package main
```

Comment:
It looks like you've got a lot of duplication here and that you're testing a client and server separately. Rather make packages `server` and `client` and a separate `integration_test` and test them together instead of using fakes. You may still need a fake for the database, but at least not for the entire server.

Reply:
Thanks for the feedback, I'll work on this